src/main/java/com/google/opensesame/user/UserData.java

```diff
@@ -0,0 +1,137 @@
+package com.google.opensesame.user;
+
+import com.google.opensesame.auth.AuthServlet;
+import com.google.opensesame.github.GitHubGetter;
+import java.io.IOException;
+import java.util.ArrayList;
+import org.kohsuke.github.GHUser;
+import org.kohsuke.github.GitHub;
+
+/**
+ * A fully fleshed out User of OpenSesame containing information sourced from GitHub. To conserve
+ * API calls, It would be best to only build UserData where neccesary. The main use of this class is
+ * to send data to the frontend as a JSON with Gson.
+ */
+public class UserData {
+  private Boolean isMentor;
+  private final ArrayList<String> interestTags;
+  private final ArrayList<String> menteeIds;
+  private final ArrayList<String> projectIds;
+  private final String bio;
+  private final String gitHubID;
+  private final String gitHubURL;
+  private final String image;
+  private final String location;
+  private final String name;
+  private final String userID;
+  private String email;
+
+  /**
+   * Create a UserData from a given UserEntity
+   *
+   * <p>Note: Emails will only ever be stored in a UserObject if the current user is authorized with
+   * the AuthServlet.
```

Comment:
We don't want to share user emails with unauthorized people to avoid spam and the misuse of our user's information by data scrapers. I'll update the Javadoc to reflect this reasoning.